app/controllers/case_requests_controller.rb

```diff
@@ -1,27 +1,66 @@
 class CaseRequestsController < ApplicationController
+  rescue_from GlimrApiClient::Case::InvalidCaseNumber, with: :case_not_found
+
   def new
-    @case_request = CaseRequest.new(nil, nil)
+    @case_request = CaseRequest.new
   end
 
   def create
-    case_reference, confirmation_code = case_request_params
-    @case_request = CaseRequest.new(case_reference, confirmation_code)
+    @case_request = CaseRequest.new(case_request_params)
 
-    if @case_request.valid?
+    respond_to do |format|
+      format.html do
+        process_html(@case_request)
+      end
+      format.json do
+        process_json(@case_request)
+      end
+    end
+  end
+
+  def show
+    @case_request = CaseRequest.find_by_id(params[:id])
+    if @case_request
       @case_request.process!
```

Comment:
This ends up creating Fee records in the database. Are we sure this should be happening in 'show'?